common/achievement/evaluate.ts

```diff
@@ -0,0 +1,126 @@
+import { Achievement_event } from '../../graphql/generated';
+import { BucketConfig, BucketEvents, ConditionDataAggregations, EvaluationResult } from './types';
+import { prisma } from '../prisma';
+import { aggregators } from './aggregator';
+import swan from '@onlabsorg/swan-js';
+import { bucketCreatorDefs } from './bucket';
+import { getLogger } from '../logger/logger';
+import { getBucketContext } from './util';
+import tracer from '../logger/tracing';
+
+const logger = getLogger('Achievement');
+
+export const evaluateAchievement = tracer.wrap('achievement.evaluateAchievement', _evaluateAchievement);
+
+async function _evaluateAchievement(
+    userId: string,
+    condition: string,
+    dataAggregation: ConditionDataAggregations,
+    recordValue: number,
+    relation?: string
+): Promise<EvaluationResult | undefined> {
+    // We only care about metrics that are used for the data aggregation
+    const metrics = Object.values(dataAggregation).map((entry) => entry.metric);
```

Comment:
Why is there achievementTemplate.metrics at all then?

Reply:
This is a very good question, which I am not able to answer. @LomyW do you know why this was done in the concept as we have it here?

Reply:
Let's try to remove it and see if something breaks. I think it's a leftover from one of our first iterations.